### ISSUE TITLE
Find/Replace All Suboutline only same as Node only

### ISSUE THREAD

**Issue by JohnLunzer** [leo-editor/leo-editor#188]
I looked into this and I found a place in the code that may explain why "Suboutline only" does not work as I expect. This function in the node find.initBatchCommands (sets in_headline):

```python
def initBatchCommands (self):
   '''Init for find-all and replace-all commands.'''
   c = self.c
   self.errors = 0
   self.in_headline = self.search_headline # Search headlines first.
   # Select the first node.
   if self.suboutline_only or self.node_only:
       self.p = c.p.copy()
   else:
       p = c.rootPosition()
       if self.reverse:
           while p and p.next():
               p = p.next()
           p = p.lastNode()
       self.p = p
   # Set the insert point.
   self.initBatchText()
```
It looks like if self.suboutline_only is true it would do the same searching (just the current node) as if I selected "Node only".

**Comment by edreamleo** [comment]
As you say, initBatchCommands sets self.p to c.p.copy() for both node-only and suboutline-only searches.  But that does not mean that those two searches are same.  The actual searching is handled elsewhere.

**Comment by edreamleo** [comment]
I am going to mark this item as invalid.  My tests pass.

Please reopen only if you can demonstrate that node-only or suboutline-only searches do not work as expected.  Note that the starting cursor position may affect all searches, which can be confusing but probably can not be helped.

**Comment by JohnLunzer** [comment]
Maybe I don't understand what suboutline means. I can give a pretty specific instance of when this doesn't work. Create this outline:
* H: 'Winged' - B: ''
  * H: Plane - B:'wheels'
  * H: Bird - B: 'feet'
  * H: Man - B: 'feet'

Make sure the 'Winged' node is selected and do a Ctrl+F replace all 'feet' to 'footers'. Nothing will get replaced. Just from looking at initBatchCommands it looks like it might be responsible for the actual traversing of the tree.

The example I gave doesn't behave as I expect it to. To me suboutline means current node, children, and children of children. If it means something else please let me know.

**Comment by JohnLunzer** [comment]
I gave that whole file a look over again and you're right initBatchCommands doesn't seem like the culprit as it looks like it's only setting the start position. However something is still amiss.

**Comment by edreamleo** [comment]
I'll reopen this issue at least until I have time to look at your example.

**Comment by edreamleo** [comment]
@JohnLunzer I do see a failure with the find-all and replace-all command, as indicated in the title of this bug. However, Ctrl-F (start-search) and Ctrl-- (replace-then-find) work as expected.

I see the following error when doing find-all or replace-all in your example:

```python
outsideSearchRange Can not happen: onlyPosition! Plane
```

This warning comes from nextNodeAfterFail, so it's clear where the problem is.  Otoh, this is an exceedingly tricky method, so the fix may not be straightforward.  Perhaps a special-case for batch find/replace will be required.

**Comment by edreamleo** [comment]
Fixed at a358e519f. All tests pass. Please reopen if problems remain.

**Comment by JohnLunzer** [comment]
Problem does not remain. Thanks.
